src/main/controllers/models/token-price.js

```diff
@@ -1,63 +1,26 @@
-const Promise = require('bluebird');
-
-module.exports = function(app, sqlLiteService) {
-	const TABLE_NAME = 'token_prices';
-	const Controller = function() {};
-
-	let knex = sqlLiteService.knex;
-
-	/**
-	 *
-	 */
-	Controller.findAll = _findAll;
-	Controller.findBySymbol = _findBySymbol;
-	Controller.add = _add;
-	Controller.edit = _edit;
-	Controller.bulkEdit = _bulkEdit;
-	Controller.bulkAdd = _bulkAdd;
-
-	function _findAll() {
-		return new Promise((resolve, reject) => {
-			knex(TABLE_NAME)
-				.select()
-				.then(rows => {
-					resolve(rows);
-				})
-				.catch(error => {
-					reject({ message: 'error_findAll', error: error });
-				});
-		});
-	}
-
-	function _findBySymbol(symbol) {
-		return new Promise((resolve, reject) => {
-			knex(TABLE_NAME)
-				.select()
-				.where({ symbol: symbol })
-				.then(rows => {
-					resolve(rows && rows.length ? rows[0] : null);
-				})
-				.catch(error => {
-					reject({ message: 'error_findAll', error: error });
-				});
-		});
-	}
-
-	function _add(tokenPrice) {
-		return sqlLiteService.insertIntoTable(TABLE_NAME, tokenPrice);
-	}
-
-	function _edit(tokenPrice) {
-		return sqlLiteService.updateById(TABLE_NAME, tokenPrice);
-	}
-
-	function _bulkEdit(tokenPrices) {
-		return sqlLiteService.bulkUpdateById(TABLE_NAME, tokenPrices);
-	}
-
-	function _bulkAdd(tokenPrices) {
-		return sqlLiteService.bulkAdd(TABLE_NAME, tokenPrices);
-	}
-
-	return Controller;
+const { knex } = require('../../services/knex');
+
+const TABLE_NAME = 'token_prices';
+
+const controller = {
+	all: () => knex(TABLE_NAME).select(),
+
+	findBySymbol: symbol =>
+		knex(TABLE_NAME)
+			.select()
+			.where({ symbol })
+			.then(rows => (rows && rows.length ? rows[0] : null)),
+
+	add: tokenPrice => knex(TABLE_NAME).insert(tokenPrice),
+
+	edit: tokenPrice =>
```

Comment:
@dfcowell Curious, is there a reason why the edit is not using transaction? Cheers

Reply:
I was in a hurry and forgot to implement transactions. 😄 This is why we do code reviews!